server/server.js

```diff
@@ -51,64 +57,95 @@ module.exports = function(__dirname) {
 	app.use(renderer.handle.bind(renderer))
 	app.use(api)
 
+	/**
+	  * Annotations
+	**/ 
+
+	var requireAdmin = (fn) =>
+		function (req, res) {
+			if (req.user) {
+				return fn(req, res)
+			} else {
+				req.status(403)
+				res.send("Please Log In first")
+			}
+		}
 
 	/**
 	  * App routing
 	**/
 
 	app.use(express.static("build"))
+	app.use("/upload", express.static(config.paths.upload))
+
+	app.post("/visible", requireAdmin(function(req, res) {
+		handleVisibility(req.body.state, req.body.page).
+			then(function() {
+				res.status(200)
+				res.send({
+					"state": req.body.state ? "visible" : "hidden",
+					"visible": req.body.state
+				})
+				renderer.reload()
+			}).
+			catch(function(err) {
+				res.status(500)
+				res.send({
+					"state": !req.body.state ? "visible" : "hidden",
+					"visible": !req.body.state
+				})
+				logger.error(err)
+			})
+	}))
 
+	app.post("/static/", upload.single('file'), requireAdmin(function(req, res) {
+		let hash = req.file.filename.slice(0,8)
+		let sluggedFile = req.file.originalname
+					.split(".")
+					.map((e) => utils.slugify(e))
+					.join(".")
+		/** No directory traversals in my house **/
+		sluggedFile = sluggedFile.replace(/\.(\.)+/g, "")
+		let prefixName = `${hash}-${sluggedFile}`
+		fs.rename(req.file.path, path.join(__dirname, config.paths.upload, prefixName)) 
+			.then( () => res.send({path: path.join("/upload/", prefixName), 
+			                       file: prefixName,
+				               shortFile: sluggedFile}) )
+			.catch( (e) => {
+				logger.error(e)
+				res.status(503).send({error: "Could not upload file"})
+			})
+	}))
 
-	app.post("/visible", function(req, res) {
-		if (req.user) {
-			handleVisibility(req.body.state, req.body.page).
-				then(function() {
-					res.status(200)
-					res.send({
-						"state": req.body.state ? "visible" : "hidden",
-						"visible": req.body.state
-					})
-					renderer.reload()
-				}).
-				catch(function(err) {
-					res.status(500)
-					res.send({
-						"state": !req.body.state ? "visible" : "hidden",
-						"visible": !req.body.state
-					})
-					logger.error(err)
-				})
-		} else {
-			req.status(403)
-			res.send("Please Log In first")
-		}
-	})
+	app.delete("/static/:FILE", requireAdmin(function(req, res) {
+		var filePath = path.join(__dirname, config.paths.upload, req.params.FILE)
+		fs.stat(filePath).then( (stat) => {
+			if (stat || stat.isFile()) {
+				//All Good!
+				return
+			} else {
+				throw new Error("File Not Found")
```

Comment:
Actually, you're not sending any response at all, if I'm reading this correctly.  It'll just spin.